pci/src/device.rs

```diff
@@ -83,18 +83,20 @@ pub trait PciDevice: BusDevice {
     /// Sets a register in the configuration space.
     /// * `reg_idx` - The index of the config register to modify.
     /// * `offset` - Offset in to the register.
-    fn write_config_register(&mut self, reg_idx: usize, offset: u64, data: &[u8]);
+    fn write_config_register(&mut self, _reg_idx: usize, _offset: u64, _data: &[u8]) {}
```

Comment:
Why do you need to implement a default implementation here?

Reply:
I think the high level trait need have a default one in case that some device as a PciDevice doesn't want to implement this function.

Reply:
This seems like it would be an exception. So I'd prefer if there would be no default for that one.

Reply:
So it means there would no case that doesn't implement this function? Umm, maybe right. Let me set back and quickly update.

Reply:
> So it means there would no case that doesn't implement this function? Ummm, right. Let me set back and quickly update.

No, it means that not implementing this method is not the default case. You typically provide a default implementation for cases where a method is most likely not implemented or common for most implementations. In our case, no implementation would be an exception and should be explicitly handled by the implementer by providing an empty of trivial implementation.

Reply:
Thanks for the explanation. I updated the codes removing the default implementation.